### ISSUE TITLE
how to use qor admin with gorm.io/gorm

### ISSUE THREAD

**Issue by hpdipto** [qor/admin#251]
When I'm trying to use qor with gorm.io/gorm, I got this error
`cannot use DB (type *"gorm.io/gorm".DB) as type *"github.com/jinzhu/gorm".DB in field value`
how can I deal with this problem?

**Comment by NilsJPWerner** [comment]
Yeah I think qor needs to be updated to support gorm 2. It seems all development of gorm has shifted to the new version.

**Comment by raven-chen** [comment]
hi @hpdipto 

What specific feature do you want to use in gorm 2.0?  We do have a plan to migrate qor to gorm 2.0, but haven't evaluated the impact and benefit. it would be helpful if you can show us your requirement.

**Comment by hpdipto** [comment]
> hi @hpdipto
> 
> What specific feature do you want to use in gorm 2.0? We do have a plan to migrate qor to gorm 2.0, but haven't evaluated the impact and benefit. it would be helpful if you can show us your requirement.

Actually I can't mention very specific feature of gorm 2.0 that I want to use. But why should I incorporate an old version in my code base?

**Comment by NilsJPWerner** [comment]
Gorm 2 has a host of new useful features like batch insert better clause support and more. We want to move to it so we get continued support and to get access to all the other features.

**Comment by KhalifaGad** [comment]
any updates?

**Comment by benjfield** [comment]
Do we have an update on this please? Gorm 1.9 has been dead for over a year and is pinned to use a version of a postgres driver that is 14 months or so older than that.

**Comment by zareef26** [comment]
Any updates?

**Comment by jpmeijers** [comment]
I'm trying to add Qor to an existing code base. My existing code has been ported to Gorm 2 a very long time ago. I was shocked to see Qor only supports the old version.

Gorm 2 has much better support for indexes in Postgres than the previous one, so there is no option of downgrading.

What is the status of porting Qor to Gorm 2, and is there perhaps a workaround like force casting the Db object to the correct type which we can do in the meantime?

**Comment by ElegantSoft** [comment]
Any updates or fork fixes this problem

**Comment by WalterNyeko** [comment]
Kind of hard to believe that there is no fix yet after 1 year.

**Comment by fanshuai** [comment]
Any updates?

**Comment by UmiKami** [comment]
![84 years](https://media0.giphy.com/media/CvHALswPAA34FjihPs/giphy.gif?cid=6c09b952t3x20s7s39occkuo1xazjkx8nhf30dcj3t4wrzy6&ep=v1_internal_gif_by_id&rid=giphy.gif)
